src/controller/CHIPDeviceController.cpp

```diff
@@ -916,12 +916,18 @@ DeviceCommissioner::ContinueCommissioningAfterDeviceAttestation(DeviceProxy * de
 CHIP_ERROR DeviceCommissioner::StopPairing(NodeId remoteDeviceId)
 {
     VerifyOrReturnError(mState == State::Initialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(remoteDeviceId != kUndefinedNodeId, CHIP_ERROR_INVALID_ARGUMENT);
+
+    bool stopped = mSetUpCodePairer.StopPairing(remoteDeviceId);
 
     CommissioneeDeviceProxy * device = FindCommissioneeDevice(remoteDeviceId);
-    VerifyOrReturnError(device != nullptr, CHIP_ERROR_INVALID_DEVICE_DESCRIPTOR);
+    if (device != nullptr)
+    {
+        ReleaseCommissioneeDevice(device);
+        stopped = true;
+    }
 
-    ReleaseCommissioneeDevice(device);
-    return CHIP_NO_ERROR;
+    return (stopped) ? CHIP_NO_ERROR : CHIP_ERROR_INVALID_DEVICE_DESCRIPTOR;
```

Comment:
So if `StopPairing` succeed but we can't find a `CommissioneeDeviceProxy` we would return `CHIP_NO_ERROR`. Is that the expected behaviour ?

Reply:
That seems reasonable: we might be doing discovery for that node id and haven't gotten far enough along yet to create the CommissioneeDeviceProxy.

Reply:
Yes, the intent seemed to be that stop returns NO_ERROR if we've actually stopped something.